src/pypromice/process/join_l3.py

```diff
@@ -0,0 +1,306 @@
+    
+    # fit = np.polyfit(df.loc[df.HS_combined.notnull(),:].index.astype('int64'),  
+    #                    df.loc[df.HS_combined.notnull(),'HS_combined'], 1)
+    # fit_fn = np.poly1d(fit)
+    
+    # df['HS_combined'] = df['HS_combined'].values \
+    #     - fit_fn(
+    #         df_in.loc[[df_in.z_surf_combined.first_valid_index()],:].index.astype('int64')[0]
+    #     )  +  df_in.loc[df_in.z_surf_combined.first_valid_index(), 'z_surf_combined']
+    # return l3_merged
+
+def build_station_list(config_folder: str, target_station_site: str) -> list:
+    """
+    Get a list of unique station IDs (stid) for a given station site.
+
+    Parameters
+    ----------
+    config_folder : str
+        Path to the folder containing the station configuration TOML files.
+    target_station_site : str
+        The station site to filter the station IDs by.
+
+    Returns
+    -------
+    list
+        A list of unique station IDs that have the specified station site.
+    """
+    unique_stids = []  # Initialize an empty list to store unique station IDs
+    
+    for filename in os.listdir(config_folder):
+        if filename.endswith(".toml"):
+            file_path = os.path.join(config_folder, filename)
+            
+            with open(file_path, 'r') as file:
+                data = toml.load(file)  # Load the TOML file
+                station_site = data.get("station_site")  # Get the station site
+                stid = data.get("stid")  # Get the station ID
+                
+                # Check if the station site matches the target and stid is unique
+                if station_site == target_station_site and stid and stid not in unique_stids:
+                    unique_stids.append(stid)  # Add the stid to the list if unique
+    
+    return unique_stids
+
+def join_l3():
+    args = parse_arguments_joinl3()
+    logging.basicConfig(
+        format="%(asctime)s; %(levelname)s; %(name)s; %(message)s",
+        level=logging.INFO,
+        stream=sys.stdout,
+    )
+    
+    # Get the list of stations associated with the given site
+    list_stations = build_station_list(args.config_folder, args.site)
+    
+    # Read the datasets and store them into a list along with their latest timestamp
+    list_station_data = []
+    for stid in list_stations:       
+        is_promice = False
+        is_gcnet = False
+        filepath = os.path.join(args.folder_l3, stid, stid+'_hour.nc')
+        if os.path.isfile(filepath):
+            is_promice = True
+        else:
+            filepath = os.path.join(args.folder_gcnet, stid+'.csv')
+            if os.path.isfile(filepath):
+                is_gcnet = True
+        if not is_promice and not is_gcnet:            
+            logger.info(stid+' not found either in '+args.folder_l3+' or '+args.folder_gcnet)
+            continue
+
+        l3, _ = loadArr(filepath)            
+        list_station_data.append((l3, l3.time.max().values, stid))
```

Comment:
Consider these minor suggestions:
* Setting the stid (key) as the first value. This makes it compatible with a `dict(list_station_data)`
* Reading the max value while sorting instead. 
```python
sorted(list_station_data, key=lambda ds: ds.time.max(), reverse=True)
```

```suggestion
        list_station_data.append((stid, l3))
```

Reply:
Updated using your suggestion here: https://github.com/GEUS-Glaciology-and-Climate/pypromice/commit/9d56d84070bb2de042412f46b265907416496680